lib/hal/xbox.c

```diff
@@ -18,6 +18,8 @@
 
 #define LaunchDataPageSize 0x1000
 
+static const char *xonline_dash_path = "\\Device\\Harddisk0\\partition2\\XODash\\xonlinedash.xbe";
```

Comment:
nit: p -> P

And is this path valid for all known online revisions?

Reply:
This is the verbatim path from reversing a devkit sample XBE. 

If someone has a collection of all of the online revisions and it turns out that they use different paths they could be looped over until a working one is found (I don't have a list to use). 

The xbe also appeared to have handling to execute a dashboard upgrade or fallback to an `XODash` path on the DVD, neither of which seem appropriate for this method since an nxdk app would presumably never have an update to apply nor a bundled `xonlinedash.xbe` to defer to.

Reply:
Capitalized the "P" and added a check to make sure xonlinedash exists. If it doesn't, this falls back to the current behavior of dropping into the dashboard.

Not sure if it'd be better to assert or just not patch up the xbe path and let the dashboard show an error since I believe the network settings are all expected to relaunch the title after completion.

Reply:
> And is this path valid for all known online revisions?

For the storage drive's path, yes. The capitalize P I had seen was from older online revisions. The latest doesn't have a capitalized letter since the file path is case-insensitive. Older online revisions had the folder named "XODASH" then later changed to "XODash". It doesn't matter to choose between the two as the kernel will be able to find the file.

I agree with abaire as there's no need to add a backup path to homebrew DVD containing `xonlinedash.xbe`.

---
NOTE: The reviewers not in XboxDev team are not able to close the review remark.